src/cctz.h

```diff
@@ -249,8 +278,77 @@ std::string Format(const std::string& format, const time_point& tp,
 //   "00.x" -> 00.x  // exact
 //
 // Errors are indicated by returning false.
+//
+// Example:
+//   const cctz::TimeZone tz = ...
+//   std::chrono::system_clock::time_point tp;
+//   if (cctz::Parse("%Y-%m-%d", "2015-10-09", tz, &tp)) {
+//     ...
+//   }
+template <typename D>
 bool Parse(const std::string& format, const std::string& input,
-           const TimeZone& tz, time_point* tpp);
+           const TimeZone& tz, time_point<D>* tpp);
+
+}  // namespace cctz
+
+//
+// IMPLEMENTATION DETAILS
+//
+namespace cctz {
+
+namespace internal {
+// Floors tp to a second boundary. Optionally returns subseconds.
+template <typename D, typename Subseconds = std::chrono::nanoseconds>
```

Comment:
Why nanoseconds?  It's arbitrary and only applies in the nullptr case, right?  D::duration perhaps?